sdk/keyvault/azure-security-keyvault/tests/test_secrets_client.py

```diff
@@ -227,52 +224,64 @@ def test_backup_restore(self, vault_client, **kwargs):
             secrets[secret_name] = client.set_secret(secret_name, secret_value)
 
         # delete all secrets
         for secret_name in secrets.keys():
             client.delete_secret(secret_name)
-
-        if not self.is_playback():
-            time.sleep(20)
+        for secret_name in secrets.keys():
+            self._poll_until_no_exception(
+                functools.partial(client.get_deleted_secret, secret_name), ResourceNotFoundError
+            )
 
         # validate all our deleted secrets are returned by list_deleted_secrets
         deleted = [s.name for s in client.list_deleted_secrets()]
-
         self.assertTrue(all(s in deleted for s in secrets.keys()))
 
         # recover select secrets
-        for secret_name in [s for s in secrets.keys() if s.startswith("secrec")]:
+        for secret_name in secrets.keys():
             client.recover_deleted_secret(secret_name)
 
-        # purge select secrets
-        for secret_name in [s for s in secrets.keys() if s.startswith("secprg")]:
-            client.purge_deleted_secret(secret_name)
+        # validate the recovered secrets exist
+        for secret_name in secrets.keys():
+            self._poll_until_no_exception(functools.partial(client.get_secret, secret_name), ResourceNotFoundError)
+
+    @ResourceGroupPreparer()
+    @VaultClientPreparer(enable_soft_delete=True)
+    def test_purge(self, vault_client, **kwargs):
+        self.assertIsNotNone(vault_client)
+        client = vault_client.secrets
 
-        if not self.is_playback():
-            time.sleep(20)
+        secrets = {}
 
-        # validate none of our purged secrets are returned by list_deleted_secrets
+        # create secrets to purge
+        for i in range(self.list_test_size):
+            secret_name = "secret{}".format(i)
+            secret_value = "value{}".format(i)
+            secrets[secret_name] = client.set_secret(secret_name, secret_value)
+
+        # delete all secrets
+        for secret_name in secrets.keys():
+            client.delete_secret(secret_name)
+        for secret_name in secrets.keys():
+            self._poll_until_no_exception(
+                functools.partial(client.get_deleted_secret, secret_name), ResourceNotFoundError
+            )
+
+        # validate all our deleted secrets are returned by list_deleted_secrets
         deleted = [s.name for s in client.list_deleted_secrets()]
-        self.assertTrue(not any(s in deleted for s in secrets.keys()))
+        self.assertTrue(all(s in deleted for s in secrets.keys()))
 
-        # validate the recovered secrets
-        expected = {k: v for k, v in secrets.items() if k.startswith("secrec")}
-        actual = {k: client.get_secret(k, "") for k in expected.keys()}
-        self.assertEqual(len(set(expected.keys()) & set(actual.keys())), len(expected))
+        # purge secrets
+        for secret_name in secrets.keys():
+            client.purge_deleted_secret(secret_name)
```

Comment:
could add a check here similar to one in [keys](https://github.com/Azure/azure-sdk-for-python/blob/1fc839b0693c326970730b41a6692ef8aa3ecc2f/sdk/keyvault/azure-security-keyvault/tests/test_key_client.py#L314) to validate that the secrets actually got deleted.